Replace dashes with underscores in lib names

phi-gamma opened the pull request:
Because of the dash in “scrypt-kdf.mllib”, compiled objects receive a
dash as well:

    ~/.opam/4.02.3/lib/scrypt-kdf/scrypt-kdf.a
    ~/.opam/4.02.3/lib/scrypt-kdf/scrypt-kdf.cma
    ~/.opam/4.02.3/lib/scrypt-kdf/scrypt-kdf.cmxa
    ~/.opam/4.02.3/lib/scrypt-kdf/scrypt-kdf.cmxs

This conflicts with the specification in META which uses underscores
throughout:

    archive(byte) = "scrypt_kdf.cma"
    archive(native) = "scrypt_kdf.cmxa"
    plugin(byte) = "scrypt_kdf.cma"
    plugin(native) = "scrypt_kdf.cmxs"
    exists_if = "scrypt_kdf.cma"

Thus findlib will ignore the installed (through OPAM) package. Replacing
the dashes with underscores fixes this.

Signed-off-by: Philipp Gesang <phg@phi-gamma.net>

phi-gamma committed:
Replace underscores with dashes in lib names

Because of the dash in “scrypt-kdf.mllib”, compiled objects receive a
dash as well:

    ~/.opam/4.02.3/lib/scrypt-kdf/scrypt-kdf.a
    ~/.opam/4.02.3/lib/scrypt-kdf/scrypt-kdf.cma
    ~/.opam/4.02.3/lib/scrypt-kdf/scrypt-kdf.cmxa
    ~/.opam/4.02.3/lib/scrypt-kdf/scrypt-kdf.cmxs

This conflicts with the specification in META which uses underscores
throughout:

    archive(byte) = "scrypt_kdf.cma"
    archive(native) = "scrypt_kdf.cmxa"
    plugin(byte) = "scrypt_kdf.cma"
    plugin(native) = "scrypt_kdf.cmxs"
    exists_if = "scrypt_kdf.cma"

Thus findlib will ignore the installed (through OPAM) package. Rename
the object files with underscores so the findlib name matches the one
used by OPAM.

Signed-off-by: Philipp Gesang <phg@phi-gamma.net>

hannesm commented:
The inverse change - using dashes in `META` instead of underscore, is IMHO a better choice.  This makes the opam and ocamlfind names the same (if they are not, I usually struggle and write the wrong one in `_tags` or `opam`) ;)

phi-gamma committed:
Export module Salsa20_core

The linker fails to resolve the module:

    Error: Error while linking /home/phg/.opam/4.02.3/lib/scrypt-kdf/scrypt-kdf.cma(Scrypt_kdf):
    Reference to undefined global `Salsa20_core'

phi-gamma commented:
2017-02-20 (Monday), notifications@github.com (Hannes Mehnert):

> The inverse change - using dashes in `META` instead of
> underscore, is IMHO a better choice.  This makes the opam and
> ocamlfind names the same (if they are not, I usually struggle and
> write the wrong one in `_tags` or `opam`) ;)

Good point actually; PR updated.

abeaumont commented:
Thanks! We'll publish a new version shortly to opam with these fixes.
